### ISSUE TITLE
Crashes and issues on MacOS

### ISSUE THREAD

**Issue by stellaraccident** [llvm/mlir-npcomp#254]
Now that #154 is resolved, we get further on MacOS, exposing a couple of other issues. Tracking them here.

**Comment by stellaraccident** [comment]
* NPE after an aligned_alloc that MacOS sometimes returns nullptr on.
* Test fragility doing a string match on something that evaluates/prints -0.0

**Comment by stellaraccident** [comment]
Need to update docs. Here was my setup procedure:

```shell
python3 -m venv ~/.venv
source ~/.venv/bin/activate

python -m pip install -r external/llvm-project/mlir/python/requirements.txt
cmake -GNinja -Bbuild -DCMAKE_BUILD_TYPE=Release -DNPCOMP_ENABLE_PYTORCH=ON .
cd build
ninja
ninja check-npcomp check-frontends-pytorch
```

With patches, the first is clean and the second has three issues that seem Torch version/platform related.

**Comment by stellaraccident** [comment]
```
export PYTHONPATH=$PWD/python_packages/npcomp_core:$PWD/python_packages/npcomp_torch
touch ../.env
../tools/torchscript_e2e_test.sh --config=refbackend
```

Needs a patch to a .so -> .dylib suffix for the refbackend runtime library. Also, Torch reports this error a lot, which crashes the test framework: `RuntimeError: Didn't find engine for operation quantized::linear_prepack NoQEngine`. Moved the offending setup code into the test exception handler to at least proceed. The result is that most integration tests pass.

**Comment by stellaraccident** [comment]
@amaleewilson This should fix your issues.

**Comment by stellaraccident** [comment]
Fixed via https://github.com/llvm/mlir-npcomp/pull/255

**Comment by amaleewilson** [comment]
It did, in fact, fix the issues! Thank you!

**Comment by silvasean** [comment]
Yeah, it looks like we can close it. We can make a new, more specific bug if any more Mac issues arise.
